### PR TITLE
Update matching_engine_utils.py to support more dynamic parameters

### LINKED ISSUE #288: typo at create_index
I have found a typo at line 95, "treeah" should be replaced with "tree-ah" =D
https://github.com/GoogleCloudPlatform/generative-ai/tree/main/language/use-cases/document-qa/utils/matching_engine_utils.py

### PR DESCRIPTION
**Pull request by dvmorris**
---
Update from @holtskinner 

Fixes #288
Replaces #300

### CODE REVIEW

File: language/use-cases/document-qa/utils/matching_engine_utils.py
Comment (RajeshThallam) [inline language/use-cases/document-qa/utils/matching_engine_utils.py:18]: To avoid breaking changes, can we make index_endpoint_name as optional and then default to `{index_name}-endpoint`.

**Review by RajeshThallam — COMMENTED** [review]
Thanks for contributing. Added a minor comment.

**Comment by dvmorris** [comment]
Yes, I think that's a great idea, sorry I didn't think about that before
submitting.

**Comment by polong-lin** [comment]
@RajeshThallam Could you do a final check and approve? I can merge after your approval

File: language/use-cases/document-qa/utils/matching_engine_utils.py
Comment (holtskinner) [reply language/use-cases/document-qa/utils/matching_engine_utils.py:18]: I made this change so this PR can be merged.

**Review by holtskinner — APPROVED** [review]

**Pull request merged**
